collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -585,6 +559,26 @@ private boolean loadSharedPreferencesFromFile(File src) {
         return res;
     }
 
+    @Override
+    public void onCorrectAdminPassword(AdminPasswordDialog.Action action) {
+        switch (action) {
+            case ADMIN_SETTINGS:
+                startActivity(new Intent(this, AdminPreferencesActivity.class));
+                break;
+            case STORAGE_MIGRATION:
+                StorageMigrationDialog storageMigrationDialog = (StorageMigrationDialog) DialogUtils.getDialogFragment(StorageMigrationDialog.class, getSupportFragmentManager());
```

Comment:
Uuuups I don't understand... `showIfNotShowing` what? A user should be asked for password when they clicks on `MIGRATE` button from `StorageMigrationDialog` not `LEARN MORE AND MIGRATE` from the `Main Menu`. Am I missing something or maybe it's you :smile:

Reply:
hah yeah sorry I meant `DialogUtils.showIfNotShowing`. Does that make sense? I wouldn't be surprised if I have the flow wrong in my head here!

Reply:
I know you meant `DialogUtils.showIfNotShowing` but there is nothing to show here.

Reply:
I guess my head it's a nice way of "fetching" while also ensuring it's definitely on screen. Here it looks like you do expect the dialog to be showing. Is that right? If so there is a case here where if the dialog for some reason **isn't** showing then this is just a dead end (as `storageMigrationDialog != null` wouldn't pass). The code could just be:

```java
case STORAGE_MIGRATION:
    DialogUtils.showIfNotShowing(new StorageMigrationDialog(), getSupportFragmentManager()).startStorageMigration();
```

If we did that we should probably change `showIfNotShowing` to take a class rather than an instance (to not waste allocations).

Reply:
Ok, now I understand what you mean, fixed. Thanks!

Reply:
Sorry I absolutely should have just put the example in at first!